cassandane/Makefile

```diff
@@ -63,18 +63,11 @@ ifndef NOCYRUS
     CYRUS_PERL_PATHS := $(shell $(PERL) utils/cyrus-perl-paths.pl)
 endif
 
-SYNTAX_rules =
+syntax: .cassandane-syntax-check
 
-define SYNTAX_template
- $(1)_syntax: $(1)
-	@$(PERL) $(CYRUS_PERL_PATHS) -c $(1)
- SYNTAX_rules += $(1)_syntax
-endef
+.cassandane-syntax-check: $(SCRIPTS) $(MODULES)
+	rm -f .cassandane-syntax-check
+	$(foreach path,$?,$(PERL) $(CYRUS_PERL_PATHS) -c $(path) || exit;)
```

Comment:
This implementation, where the foreach is inside a rule, serialises the checks.  We can no longer benefit from parallel make (`make -j4` or whatever).

The purpose of the odd-looking `SYNTAX_template` design, where the foreach calls define rules rather than performing actions, is so that we end up with one rule per file.  This means a parallel make can run a bunch of the checks at a time.  This is a significant speedup vs running them serially:

```
ellie@debian:master:~/fastmail/cyrus-imapd/cassandane$time make
...
real	0m54.897s
user	0m47.373s
sys	0m6.841s
ellie@debian:master:~/fastmail/cyrus-imapd/cassandane$time make -j4
...
real	0m32.507s
user	0m51.438s
sys	0m8.984s
```

Reply:
Yeah I discovered that a bit later :)